### PR TITLE
Handle content store 404s

### PR DESCRIPTION
**Pull request by theleebriggs**
# Trello
https://trello.com/c/wyB5V6m0/108-a-content-item-on-gov-uk-is-causing-import-task-import-content-items-by-organisation-to-fail

# Description
Some content items found using the search api may not have a corresponding entry in the content store API - these will return a 404 which in turn will stop the import of an organisations content items at that point.

This PR adds 404 logic to the content store service as well as an overhaul of the http request mocking in the test suite, adding in WebMock rather than hijacking HTTParty. By doing this the tests remain http client agnostic.

### CODE REVIEW

File: app/services/clients/content_store.rb [hunk 1]
```diff
@@ -3,9 +3,13 @@ class ContentStore
...
         response = HTTParty.get(end_point(path))
-        content_item = JSON.parse(response.body).symbolize_keys
-
-        content_item.slice(*attributes)
+        case response.code
+          when 200
+            content_item = JSON.parse(response.body).symbolize_keys
+            content_item.slice(*attributes)
+          when 404
+            false
```
Comment (pmanrubia) [inline app/services/clients/content_store.rb]: I would prefer to return `nil` here. It feels a [bit more idiomatic to Ruby](https://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-find).

File: app/services/content_items_service.rb [hunk 2]
```diff
@@ -8,7 +8,10 @@ def find_each(organisation_slug)
     Clients::SearchAPI.find_each(query: query, fields: fields) do |response|
       base_path = response.fetch(:link)
 
-      yield Clients::ContentStore.find(base_path, attribute_names)
+      item = Clients::ContentStore.find(base_path, attribute_names)
+      if item
```
Comment (pmanrubia) [inline app/services/content_items_service.rb]: what about:

```ruby
content_item = Clients::ContentStore.find(base_path, attribute_names)

yield content_item if content_item
```

File: spec/services/content_items_service_spec.rb [hunk 3]
```diff
@@ -26,6 +26,16 @@
       expect(result).to match_array([:a])
     end
 
+    it "doesn't yield empty content items" do
```
Comment (pmanrubia) [inline spec/services/content_items_service_spec.rb]: This is a good test 👍  Would you mind updating the copy to something that make it explicit that it is not yielding `nil` responses from the `ContentStore`

File: app/services/clients/content_store.rb (hunk 1 above)
Comment (theleebriggs) [inline app/services/clients/content_store.rb]: Addressed now.

File: app/services/content_items_service.rb (hunk 2 above)
Comment (theleebriggs) [inline app/services/content_items_service.rb]: Addressed now.

File: spec/services/content_items_service_spec.rb (hunk 3 above)
Comment (theleebriggs) [inline spec/services/content_items_service_spec.rb]: Changed to "does not yield nil responses from the content store".

**Pull request merged**